### ISSUE TITLE
issues with MediaPlayer looping

### ISSUE THREAD

**Issue by craftycram** [aiortc/aiortc#918]
Hello everyone,  

I currently struggle with looping video files with the aiortc MediaPlayer helper.  
I'm using the janus example code with the loop parameter added to the MediaPlayer parameters.  
I've read that the file I want to loop has to be seekable. But I'm not sure on how to test this.  

I've tried playing videos from multiple sources. Encoded videos myself with ffmpeg, Adobe Media Encoder, YouTube online converter or just a screen recording. Whatever I try doesnt seem to work. After the video played I get the log output and it doesn't loop: `DEBUG:aiortc.contrib.media:MediaPlayer(/path/to/video.mp4) Stopping worker thread`

I randomly found [one video file](https://www.heimkino-atmos.de/wp-content/uploads/2021/06/Dolby.mp4) that works flawless. It just plays on loop endlessly.

Already tried using mp4box or ffmpeg to maybe add some missing metadata, but that doesn't seem to make any difference as well.

Does anyone know what I'm doing wrong or how I can fix my video files to work?

**Comment by DoubleBiscuit** [comment]
Looping video files using the aiortc MediaPlayer can indeed be a bit tricky, especially if the video file is not behaving as expected. Here are some steps and tips to help you troubleshoot and potentially fix the issue:

• Check Video File: Ensure that the video file you're using is indeed seekable and correctly encoded. You can check this using a tool like ffprobe (from FFmpeg) or ffinfo to inspect the file's attributes and seekability. Look for the "seekable" attribute in the file information.

bashCopy code

ffprobe -i /path/to/video.mp4 

If it's not seekable, you may need to re-encode the video to make it seekable. This often involves converting it to a format that supports seeking, like an MP4 file with H.264 video and AAC audio.

• Video Metadata: Ensure that the video file has the necessary metadata for looping. Some players rely on specific metadata tags to know whether a video should loop. You can use FFmpeg to set the loop flag in the video file metadata:

bashCopy code

ffmpeg -i /path/to/video.mp4 -c:v copy -c:a copy -movflags +faststart+rtphint -metadata:s:v:0 loop=0 /path/to/output.mp4 

This command sets the loop flag to "0," which should indicate an infinite loop.

• Video Encoding: Make sure the video file is encoded with compatible codecs. H.264 for video and AAC for audio are widely supported codecs for web playback. Ensure that the video file you're using meets these criteria.

• File Source: If some videos work while others don't, it could be related to how those videos were created or encoded. Try to identify patterns among the videos that work versus those that don't. For example, check if there are differences in encoding settings, frame rates, or resolutions.

• Video Containers: Pay attention to the video container format (e.g., MP4, MKV, AVI). Different containers may have different behaviors, and some may not be fully supported by the aiortc MediaPlayer. Stick to widely supported container formats like MP4.

• Video Bitrate: High bitrate videos can sometimes cause playback issues. Consider lowering the video bitrate during encoding to see if it resolves the looping problem.

• Library Updates: Ensure that you are using the latest version of the aiortc library and that you have applied any relevant updates or patches that address media playback issues.

• Test on Different Platforms: Try running your code on different platforms and browsers to see if the issue persists across all of them. Compatibility can vary, so testing on multiple setups can help identify specific issues.

• Debugging: Use debugging tools to inspect the state of your application and the MediaPlayer object during video playback. This can help pinpoint any errors or unexpected behavior.

• Community Support: Check the aiortc community or forums for discussions related to looping video with aiortc MediaPlayer. Others may have encountered similar issues and found solutions.

By following these steps and experimenting with different settings and encoding options, you should be able to identify and address the issue with looping video files in your aiortc application.

**Comment by craftycram** [comment]
Got it working. :rocket:

I checked the files with ffprobe. Both don't include `seekable` in their metadata, even the working video doesnt.  
Reencoding with ffmpeg with your command and other ones I tried also didn't work.  

Both videos are encoded with the same video and audio encoding.  
But I found small differences in the encoder specific settings which I tried to match with MediaEncoder which solved my issue.

Only difference I can think of changing in MediaEncoder is setting the codec profile to Baseline.  
Hope this helps if anyone else encounters this issue.
My exact settings can be found [here](https://pastebin.com/raw/iavYQiQB)
